mmhuman3d/core/visualization/renderer/torch3d_renderer/mesh_renderer.py

```diff
@@ -0,0 +1,87 @@
+from typing import Iterable, Optional, Tuple, Union
+
+import torch
+from pytorch3d.structures import Meshes
+
+from mmhuman3d.core.cameras import MMCamerasBase
+from mmhuman3d.core.visualization.renderer.torch3d_renderer.lights import \
+    MMLights  # noqa: E501
+from .base_renderer import BaseRenderer
+from .builder import RENDERER
+
+
+@RENDERER.register_module(
+    name=['Mesh', 'mesh', 'mesh_renderer', 'MeshRenderer'])
+class MeshRenderer(BaseRenderer):
+    """Render RGBA image with the help of camera system."""
+    shader_type = 'SoftPhongShader'
+
+    def __init__(
+        self,
+        resolution: Tuple[int, int] = None,
+        device: Union[torch.device, str] = 'cpu',
+        output_path: Optional[str] = None,
+        out_img_format: str = '%06d.png',
+        **kwargs,
+    ) -> None:
+        """Renderer for RGBA image of meshes.
+
+        Args:
+            resolution (Iterable[int]):
+                (width, height) of the rendered images resolution.
+            device (Union[torch.device, str], optional):
+                You can pass a str or torch.device for cpu or gpu render.
+                Defaults to 'cpu'.
+            output_path (Optional[str], optional):
+                Output path of the video or images to be saved.
+                Defaults to None.
+            out_img_format (str, optional): The image format string for
+                saving the images.
+                Defaults to '%06d.png'.
+        """
+        super().__init__(
+            resolution=resolution,
+            device=device,
+            output_path=output_path,
+            out_img_format=out_img_format,
+            **kwargs)
+
+    def forward(self,
+                meshes: Optional[Meshes] = None,
+                cameras: Optional[MMCamerasBase] = None,
+                lights: Optional[MMLights] = None,
+                indexes: Optional[Iterable[int]] = None,
+                backgrounds: Optional[torch.Tensor] = None,
+                **kwargs) -> Union[torch.Tensor, None]:
+        """Render Meshes.
+
+        Args:
+            meshes (Optional[Meshes], optional): meshes to be rendered.
+                Defaults to None.
+            cameras (Optional[MMCamerasBase], optional): cameras for render.
+                Defaults to None.
+            lights (Optional[MMLights], optional): lights for render.
+                Defaults to None.
+            indexes (Optional[Iterable[int]], optional): indexes for images.
+                Defaults to None.
+            backgrounds (Optional[torch.Tensor], optional): background images.
+                Defaults to None.
+
+        Returns:
+            Union[torch.Tensor, None]: return tensor or None.
+        """
+
+        meshes = meshes.to(self.device)
```

Comment:
It seems that `meshes` is a necessary argument, not optional. We should remove the default `None` and Optional in args list.

Reply:
ok